Dependencies/napi/source/js_native_api_JavaScriptCore.cc

```diff
@@ -10,70 +10,111 @@
 #include <algorithm>
 #include <cassert>
 
-struct FunctionTableEntry {
-  napi_callback cb;
-  void* data;
-};
-
-struct FunctionTableProperty {
-  napi_callback getter;
-  napi_callback setter;
-  void* data;
-};
-
-struct FunctionTable {
-  napi_env env;
-  std::vector<FunctionTableEntry> table;
-  std::map<std::string, FunctionTableProperty> properties;
-};
-
-struct ClassTable {
-  napi_callback cb;
-  void* data;
-  JSClassRef classRef;
-  napi_env env;
-  FunctionTable FunctionTable;
-};
-
-struct CallbackInfo {
-  napi_value newTarget;
-  napi_value thisArg;
-  napi_value* argv;
-  void* data;
-  uint16_t argc;
-  bool isConstructCall{ false };
-};
-
-struct RefInfo {
-  JSValueRef value;
-  uint32_t count;
-};
-
-std::map<JSObjectRef, ClassTable*> constructorCB;
-std::map<JSValueRef, FunctionTable*> FunctionTables;
-
-void DumpException(napi_env env, JSValueRef exception) {
-  char errorStr[1024];
-  size_t errorStrSize{ 0 };
-
-  if (!JSValueIsNull(env->m_globalContext, exception)) {
-    // Getting error string
-    if (JSValueIsString(env->m_globalContext, exception)) {
-      OpaqueJSValue* ncstr = const_cast<OpaqueJSValue*>(exception);
-      /*size_t length =*/ JSStringGetUTF8CString(reinterpret_cast<JSStringRef>(ncstr), errorStr, sizeof(errorStr));
-    } else {
-      OpaqueJSValue *exceptionValue = const_cast<OpaqueJSValue*>(exception);
-      napi_get_value_string_utf8(env, reinterpret_cast<napi_value>(exceptionValue), errorStr, sizeof(errorStr), &errorStrSize);
+namespace {
+  struct FunctionTableEntry {
+    napi_callback cb;
+    void* data;
+  };
+
+  struct FunctionTableProperty {
+    napi_callback getter;
+    napi_callback setter;
+    void* data;
+  };
+
+  struct FunctionTable {
+    napi_env env;
+    std::vector<FunctionTableEntry> table;
+    std::map<std::string, FunctionTableProperty> properties;
+  };
+
+  struct ClassTable {
+    napi_callback cb;
+    void* data;
+    JSClassRef classRef;
+    napi_env env;
+    FunctionTable FunctionTable;
```

Comment:
Rename FunctionTable to functionTable, it doesn't compile with gcc without -permissive and it's confusing.